### ISSUE TITLE
Add clarity around ReadWriteMany and CSI volume fsgroup ownership changes

### ISSUE THREAD

**Issue by pwschuurman** [kubernetes/website#46688]
**This is a Feature Request**

**What would you like to be added**

Enhance the documentation around volume permissions and ownership change policy for Pods. In the [documentation](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#configure-volume-permission-and-ownership-change-policy-for-pods) the `fsgroup` setting is not explicitly defined as only supporting `ReadWriteOnce` volumes. In the [CSI mounter code](https://github.com/kubernetes/kubernetes/blob/9e2075b3c87061d25759b0ad112266f03601afd8/pkg/volume/csi/csi_mounter.go#L475), `fsgroup` is ignored if the AccessMode for a PVC is not `ReadWriteOnce`. This request is to enhance the documentation by adding a note stating this setting is ignored in RWX mode.

**Why is this needed**
This would clarify the use of the `fsgroup` setting fpr RWX volumes

**Comments**

**Comment by mattcary** [comment]
If that ftn returns false, then I think we fall back to the kubelet fsgroup policy which will apply it for RWX volumes: https://github.com/kubernetes/kubernetes/blob/9e2075b3c87061d25759b0ad112266f03601afd8/pkg/volume/csi/csi_mounter.go#L335

?

**Comment by pwschuurman** [comment]
I think it's driver dependent. Fsgroup will only be honoured for RWX volumes if the driver supports [`VOLUME_MOUNT_GROUP`](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#delegating-volume-permission-and-ownership-change-to-csi-driver). By default the Kubelet fsgroup will only be applied if that function (`supportsFSGroup`) returns true (and the function returns false if the PVC isn't RWO).

**Comment by mattcary** [comment]
Hmm. I thought I'd seen problems for fsgroup on nfs volumes, though.

**Comment by mattcary** [comment]
Oh, but if a driver advertises, eg, a File access mode then fsgroup will apply: https://kubernetes-csi.github.io/docs/support-fsgroup.html.

**Comment by pwschuurman** [comment]
> Oh, but if a driver advertises, eg, a File access mode then fsgroup will apply: https://kubernetes-csi.github.io/docs/support-fsgroup.html.

Yes, I think that's the only way that non-RWO volumes can support fsgroup, by delegating to a (supported) CSI driver
